lib/ruby_lsp/requests/support/highlight_target.rb

```diff
@@ -0,0 +1,85 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
+  module Requests
+    module Support
+      class HighlightTarget
+        extend T::Sig
+
+        READ = LanguageServer::Protocol::Constant::DocumentHighlightKind::READ
+        WRITE = LanguageServer::Protocol::Constant::DocumentHighlightKind::WRITE
+
+        class HighlightMatch < T::Struct
+          const :type, Integer
+          const :node, SyntaxTree::Node
+        end
+
+        sig { params(node: SyntaxTree::Node).void }
+        def initialize(node)
+          @node = node
+          @value = T.let(value(node), T.nilable(String))
+        end
+
+        sig { params(other: SyntaxTree::Node).returns(T.nilable(HighlightMatch)) }
+        def highlight_type(other)
+          matched_highlight(other) if other.is_a?(SyntaxTree::Params) || @value == value(other)
+        end
+
+        private
+
+        # Match the target type (where the cursor is positioned) with the `other` type (the node we're currently
+        # visiting)
+        sig { params(other: SyntaxTree::Node).returns(T.nilable(HighlightMatch)) }
+        def matched_highlight(other)
+          case @node
+          when SyntaxTree::VCall, SyntaxTree::FCall, SyntaxTree::Call, SyntaxTree::Command,
```

Comment:
Is it possible to categorize these node types in the `when` statement, Like moving them into a descriptive constant? It'll increase readability a lot.
I know that it may not be possible because of some edge cases though.